### ISSUE TITLE
[jsk_topic_tools] Why roscpp_tutorials is listed as build_depend?

### ISSUE THREAD

**Issue by wkentaro** [jsk-ros-pkg/jsk_common#1230]
@k-okada 
Why jsk_topic_tools depends on roscpp_tutorials?
We can not test general tools in jsk_topic_tools by changing programs in roscpp_tutorials because of that.

I found this change 9cd6f44a9dc62d8a79078ed0c686b95c2c0ac324

**Comment by k-okada** [comment]
https://github.com/jsk-ros-pkg/jsk_common/pull/279/files

i think we used roscpp_tutorials at somewhere in test code


--
◉ Kei Okada

**Comment by k-okada** [comment]
very hard to follow file history but may be originated at
https://github.com/jsk-ros-pkg/jsk_common/commit/8248415f156abb8e78fe0e613d2f006ebe71443d


--
◉ Kei Okada

**Comment by k-okada** [comment]
and still used in
https://github.com/jsk-ros-pkg/jsk_common/blob/master/jsk_topic_tools/test/test_topic_buffer_close_wait.test#L8

--
◉ Kei Okada

**Comment by k-okada** [comment]
that should be in test_depend but as far as I know ,rosdep did not install
package in test_depend


--
◉ Kei Okada

**Comment by wkentaro** [comment]
That rosdep behavior seems changed at REP-0140.
https://github.com/ros-infrastructure/rep/pull/59/files#diff-3fa963019eccb75f6893ad65b7317a69R46

So currently, test_depend may work as we expect.

On Thu, Nov 12, 2015 at 11:52 AM, Kei Okada <notifications@github.com>
wrote:

> that should be in test_depend but as far as I know ,rosdep did not install
> package in test_depend
>
>
> --
> ◉ Kei Okada
>
>
> On Thu, Nov 12, 2015 at 11:49 AM, Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
>
> wrote:
>
> > and still used in
> >
> https://github.com/jsk-ros-pkg/jsk_common/blob/master/jsk_topic_tools/test/test_topic_buffer_close_wait.test#L8
> >
> > --
> > ◉ Kei Okada
> >
> >
> > On Thu, Nov 12, 2015 at 11:45 AM, Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp
> >
> > wrote:
> >
> >> very hard to follow file history but may be originated at
> >>
> https://github.com/jsk-ros-pkg/jsk_common/commit/8248415f156abb8e78fe0e613d2f006ebe71443d
> >>
> >>
> >> --
> >> ◉ Kei Okada
> >>
> >>
> >> On Thu, Nov 12, 2015 at 11:42 AM, Kei Okada <
> k-okada@jsk.t.u-tokyo.ac.jp>
> >> wrote:
> >>
> >>> https://github.com/jsk-ros-pkg/jsk_common/pull/279/files
> >>>
> >>> i think we used roscpp_tutorials at somewhere in test code
> >>>
> >>>
> >>> --
> >>> ◉ Kei Okada
> >>>
> >>>
> >>> On Thu, Nov 12, 2015 at 12:33 AM, Kentaro Wada <
> notifications@github.com
> >>> > wrote:
> >>>
> >>>> @k-okada <https://github.com/k-okada>
> >>>> Why jsk_topic_tools depends on roscpp_tutorials?
> >>>> We can not test general tools in jsk_topic_tools by changing programs
> >>>> in roscpp_tutorials because of that.
> >>>>
> >>>> I found this change 9cd6f44
> >>>> <
> https://github.com/jsk-ros-pkg/jsk_common/commit/9cd6f44a9dc62d8a79078ed0c686b95c2c0ac324
> >
> >>>>
> >>>> —
> >>>> Reply to this email directly or view it on GitHub
> >>>> <https://github.com/jsk-ros-pkg/jsk_common/issues/1230>.
> >>>>
> >>>
> >>>
> >>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_common/issues/1230#issuecomment-155982182>
> .
>



-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com
